src/domain/wallets/tx-history.ts

```diff
@@ -64,77 +66,66 @@ export const fromLedger = (
         credit,
         username,
       })
-      if (type === LedgerTransactionType.OnchainIntraLedger && addresses) {
+      if (addresses && addresses.length > 0) {
         return {
           id,
-          settlementVia: SettlementMethod.IntraLedger,
-          old: {
+          initiationVia: PaymentInitiationMethod.OnChain,
+          settlementVia:
+            type === LedgerTransactionType.OnchainIntraLedger
+              ? SettlementMethod.IntraLedger
+              : SettlementMethod.OnChain,
+          addresses,
+          deprecated: {
             description,
             usd,
             feeUsd,
             type,
           },
+          recipientId: username || null,
           settlementAmount,
-          settlementFee: fee,
-          paymentHash: null,
-          addresses: addresses,
-          recipientId: null,
+          settlementFee: toSats(fee || 0),
           pendingConfirmation,
           createdAt: timestamp,
         }
       }
-      if (type === LedgerTransactionType.IntraLedger) {
+      if (paymentHash) {
```

Comment:
There was some logic on the `main` branch that suggested that `Intraledger` payments could have `paymentHash` values as well and that this was used as the unique identifier for the `invoiceusers` table (see [this test line](https://github.com/GaloyMoney/galoy/blob/548ac0fc673280033fef8f7bff3bbb68e7401baf/test/integration/02-user-wallet/02-send-lightning.spec.ts#L68) for e.g.).

If we changed that behavior then this is ok, but if not then this could potentially give us `Intraledger` transactions as well as `Lightning` transactions I think.

Reply:
Oh nevermind, I just caught the rest of logic just under this... LGTM :+1:

Reply:
yes that is why there is a switch on the `settlementVia` field:
```
      if (paymentHash) {
        return {
          id,
          initiationVia: PaymentInitiationMethod.Lightning,
          settlementVia:
            type === LedgerTransactionType.IntraLedger
              ? SettlementMethod.IntraLedger
              : SettlementMethod.Lightning,
```